lib/rewards-bamboohr.rb

```diff
@@ -10,38 +10,42 @@ module RewardsBamboohr
   class Base
     DEFAULT_BASE_URL = "http://rewards.flts.tk/api/v1".freeze
 
-    def self.create_birthday_bonus
-      new.create_birthday_bonus
-    end
+    attr_reader :emails
 
-    def self.create_anniversary_bonus
-      new.create_anniversary_bonus
-    end
+    def initialize(emails = nil)
+      @emails = emails
 
-    def initialize
       Rewards::Client.base_uri(base_uri)
     end
 
     def create_birthday_bonus
-      birthdays = bamboohr_fetcher.today_birthdays
+      return unless birthdays
 
       Bonus.new(
         token: token,
         template: RewardsBamboohr::Templates::Birthday
-      ).create_bonuses(birthdays) if birthdays
+      ).create_bonuses(birthdays)
     end
 
     def create_anniversary_bonus
-      anniversaries = bamboohr_fetcher.today_anniversaries
+      return unless anniversaries
 
       Bonus.new(
         token: token,
         template: RewardsBamboohr::Templates::Anniversary
-      ).create_bonuses(anniversaries) if anniversaries
+      ).create_bonuses(anniversaries)
     end
 
     private
 
+    def birthdays
+      @birthdays ||= emails ? bamboohr_fetcher.by_emails(emails) : bamboohr_fetcher.today_birthdays
```

Comment:
Do you think we can refactor this way?

```suggestion
      @birthdays ||= begin
        birthdays = bamboohr_fetcher.by_emails(emails) if emails.present?
        birthdays = bamboohr_fetcher.today_birthdays       
      end
```

Reply:
It will always return the second row, won't it?
```birthdays = bamboohr_fetcher.today_birthdays```
Shouldn't we write instead of it 
```birthdays = bamboohr_fetcher.today_birthdays unless emails.present?```
?

Reply:
@RamilyaNigmatullina you are right, good catch! Do we really need memoization here at all, coult we remove it?

```ruby
def birthdays
  return bamboohr_fetcher.by_emails(emails) if emails.present?
  bamboohr_fetcher.today_birthdays
end
```

Reply:
We need memoization because it will call `birthdays` method twice if birthdays are present.